File: src/main/java/de/hannisoft/gaveplan/excelimport/GraveSiteCriteriaFileReader.java
package de.hannisoft.graveplan.excelimport;

import java.io.File;
import java.io.IOException;
import java.util.Map;

import de.hannisoft.graveplan.model.GraveSite;
import jxl.Sheet;
import jxl.Workbook;
import jxl.WorkbookSettings;
import jxl.read.biff.BiffException;

public class GraveSiteCriteriaFileReader extends AbstractXLSFileReader {
    private static final String COL_FIELD = "Feld";
    private static final String COL_ROW = "Reihe";
    private static final String COL_PLACE = "Grabstätte";
    private static final String COL_CRIT_1 = "Auswahlkriterium 1";
    private static final String COL_CRIT_2 = "Auswahlkriterium 2";
    private static final String COL_CRIT_3 = "Auswahlkriterium 3";

    public void read(String inputFile, Map<String, GraveSite> graves) throws IOException {
        File inputWorkbook = new File(inputFile);
        Workbook w;
        WorkbookSettings ws = new WorkbookSettings();
        ws.setEncoding("Cp1252");
        try {
            w = Workbook.getWorkbook(inputWorkbook, ws);
            Sheet sheet = w.getSheet(0);
            initColumnMap(sheet, 0);
            for (int i = 2; i < sheet.getRows(); i++) {
                String crit1 = getContent(COL_CRIT_1, i);
                String crit2 = getContent(COL_CRIT_2, i);
                String crit3 = getContent(COL_CRIT_3, i);

                if ((crit1 == null || crit1.trim().isEmpty()) && (crit2 == null || crit2.trim().isEmpty())
                        && (crit3 == null || crit3.trim().isEmpty())) {
                    continue;
                }

                String field = getContent(COL_FIELD, i);
                String row = getContent(COL_ROW, i);
                String place = getContent(COL_PLACE, i);
                String graveId = GraveSite.createId(field, row, place);
                GraveSite grave = graves.get(graveId);
                if (grave != null) {
                    grave.addCriteria(crit1);
                    grave.addCriteria(crit2);
                    grave.addCriteria(crit3);

                }
            }
        } catch (BiffException e) {
            e.printStackTrace();
        }
    }

}

File: src/main/kotlin/de/hannisoft/graveplan/excelimport/GraveSiteCriteriaFileReader.kt
package de.hannisoft.graveplan.excelimport

import de.hannisoft.de.hannisoft.graveplan.excelimport.buildHeaderMap
import de.hannisoft.de.hannisoft.graveplan.excelimport.getValue
import de.hannisoft.graveplan.model.GraveSite
import org.apache.poi.ss.usermodel.WorkbookFactory
import java.io.File

class GraveSiteCriteriaFileReader {
    companion object {
        const val COL_FIELD: String = "Feld"
        const val COL_ROW: String = "Reihe"
        const val COL_PLACE: String = "Grabstätte"
        const val COL_CRIT_1: String = "Auswahlkriterium 1"
        const val COL_CRIT_2: String = "Auswahlkriterium 2"
        const val COL_CRIT_3: String = "Auswahlkriterium 3"
    }

    fun read(inputFile: String, graves: Map<String, GraveSite>) {
        val sourceFile = File(inputFile)

        sourceFile.inputStream().use { fis ->
            val workbook = WorkbookFactory.create(fis)
            val sheet = workbook.getSheetAt(0)
            val headerMap = sheet.buildHeaderMap()
            for (rowIndex in 1..sheet.lastRowNum) {
                val row = sheet.getRow(rowIndex) ?: continue
                val crit1 = row.getValue(headerMap[COL_CRIT_1])
                val crit2 = row.getValue(headerMap[COL_CRIT_2])
                val crit3 = row.getValue(headerMap[COL_CRIT_3])
                if (crit1.isEmpty() && crit2.isEmpty() && crit3.isEmpty()) {
                    continue
                }

                val field = row.getValue(headerMap[COL_FIELD])
                val rowStr = row.getValue(headerMap[COL_ROW])
                val placeStr = row.getValue(headerMap[COL_PLACE])
                val graveId = GraveSite.createId(field, rowStr, placeStr)
                graves[graveId]?.apply {
                    addCriteria(crit1)
                    addCriteria(crit2)
                    addCriteria(crit3)
                }
            }
        }
    }
}